### PR TITLE
Add SceneMetadata to missing locations

### PR DESCRIPTION
**Pull request by CamronStaley**
## What changes are proposed in this pull request?

Adding sceneMetadata to public and to get_metadata_cls

```
import fiftyone as fo

dataset = fo.load_dataset("quickstart-3d")
sample = dataset.first()
filepath = sample["filepath"]
size = fo.SceneMetadata.build_for(filepath, _cache=None).size_bytes
print(size)
print(filepath)
```

## How is this patch tested? If it is not, please explain why.

ipython

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Introduced `SceneMetadata` to enhance functionality related to scene-specific data management.
  - Expanded metadata handling to support three-dimensional media types.
  - Added Elasticsearch integration for enhanced vector search capabilities.
  - New backend option for `compute_similarity()` method using Elasticsearch.

- **Documentation**
  - Updated documentation to include information about Elasticsearch integration and its setup.
  - Enhanced user guide with details on Elasticsearch as a backend option for similarity computations.

### CODE REVIEW

**Review by brimoor — CHANGES_REQUESTED** [review]
@CamronStaley nice find! Can you rebase onto `release/v0.25.0`?

**Comment by swheaton** [comment]
Lgtm. Though I don't think your test code in the PR exercises the getmetadata_cls change

**Comment by CamronStaley** [comment]
I am not exactly sure how topher's commit got in there

File: fiftyone/__public__.py [hunk 1]
```diff
@@ -113,6 +113,7 @@
     Metadata,
     ImageMetadata,
     VideoMetadata,
+    SceneMetadata,
```
[the remark this answers is not in the document]
Comment (swheaton) [reply fiftyone/__public__.py:116]: @coderabbitai In this file we import names that we want to be available to users at the top module level. Please don't make this unused import suggestion on this file from now on.

**Review by swheaton — APPROVED** [review]
lgtm

**Pull request merged**